### PR TITLE
Vendor stdbool.h from clang 15.0.2

### PR DESCRIPTION
**Pull request by cole-miller**
This follows the example of the existing headers in `resources`, copying stdbool.h from the latest tagged release of LLVM.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Comment by fbs** [comment]
o/ thanks for the contribution. Can you explain what kind of problem the vendoring solves?

**Comment by cole-miller** [comment]
Ah, I should've linked the discussion: https://github.com/iovisor/bpftrace/discussions/2378. Hopefully @viktormalik can confirm that I'm on the right track here.

**Review by viktormalik — COMMENTED** [review]
Thanks! This is useful when including headers that include `<stdbool.h>`. Could you perhaps add a runtime test with such a header?

**Comment by viktormalik** [comment]
ping @cole-miller, I'd like to see a runtime test for this before merging it

**Comment by fbs** [comment]
shall we just merge this one? it looks ok

**Comment by viktormalik** [comment]
yeah, should be fine, let's merge

**Review by viktormalik — APPROVED** [review]
I resolved the CHANGELOG merge conflict and will merge once the tests pass.

**Pull request merged**
